src/lib/stories.js

```diff
@@ -55,7 +64,62 @@ const searchStories = async (program) => {
     return stories;
 };
 
-const filterStories = ({ program, stories, projectsById, statesById, membersById, epicsById }) => {
+const hydrateStory = (entities, story) => {
+    debug('hydrating story');
+    story.project = entities.projectsById[story.project_id];
+    story.state = entities.statesById[story.workflow_state_id];
+    story.epic = entities.epicsById[story.epic_id];
+    story.owners = story.owner_ids.map(id => entities.membersById[id]);
+    debug('hydrated story');
+    return story;
+};
+
+const findProject = (entities, project) => {
+    if (entities.projectsById[project]) {
+        return entities.statesById[project];
+    }
+    const projectMatch = new RegExp(project, 'i');
+    return Object.values(entities.projectsById).filter(s => !!s.name.match(projectMatch))[0];
+};
+
+const findState = (entities, state) => {
+    if (entities.statesById[state]) {
+        return entities.statesById[state];
+    }
+    const stateMatch = new RegExp(state, 'i');
+    // Since the name of a state may be duplicated, it would be
+    // much safer to search for states of the current story workflow.
+    // That will take a bit of refactoring.
+    return Object.values(entities.statesById)
+        .filter(s => !!s.name.match(stateMatch))[0];
+};
+
+const findEpic = (entities, epicName) => {
+    if (entities.epicsById[epicName]) {
+        return entities.epicsById[epicName];
+    }
+    const epicMatch = new RegExp(epicName, 'i');
+    return Object.values(entities.epicsById)
+        .filter(s => s.name.match(epicMatch))[0];
+};
+
+const findOwners = (entities, owners) => {
```

Comment:
It would probably be better if this function name indicated that it only returns an array of owner IDs (not objects), unlike `findEpic` or `findState` that return the full entity.

Ditto with the `findLabels` below which returns only the names...